docs/UPGRADING.md

```diff
@@ -2,6 +2,394 @@
 
 Here you can find a list of migration guides to handle breaking changes between releases of the CLI.
 
+## 0.18.0
+
+### Breaking changes in gRPC API and CLI JSON output.
+
+Starting from this release we applied a more rigorous and stricter naming conventions in gRPC API following the official
+guidelines: https://developers.google.com/protocol-buffers/docs/style
+
+We also started using a linter to implement checks for gRPC API style errors.
+
+This provides a better consistency and higher quality API but inevitably introduces breaking changes.
+
+### gRPC API breaking changes
```

Comment:
```suggestion
#### gRPC API breaking changes
```
I suggest increasing the heading level to H4 since this is a sub-section of the H3 "Breaking changes in gRPC API and CLI JSON output." heading, which results in a ToC structure like:

- 0.18.0
  - Breaking changes in gRPC API and CLI JSON output.
    - gRPC API breaking changes
    - arduino-cli JSON output breaking changes
  - (any other entries for 0.18.0)